src/sparseml/pytorch/sparsification/pruning/modifier_pruning_mfac.py

```diff
@@ -0,0 +1,1469 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Modifier classes implementing M-FAC pruning as described in
+https://arxiv.org/pdf/2107.03356.pdf
+"""
+from typing import (
+    Any,
+    Dict,
+    List,
+    Optional,
+    Union,
+)
+import os
+import logging
+import math
+from dataclasses import dataclass, field
+from abc import ABC, abstractmethod
+from functools import wraps
+import GPUtil
+
+import torch
+from torch import Tensor
+from torch.nn import Module, Parameter
+import torch.distributed as dist
+from torch.nn.parallel.parallel_apply import parallel_apply
+
+from sparseml.pytorch.utils.logger import BaseLogger
+from sparseml.pytorch.utils import GradSampler
+from sparseml.pytorch.optim.modifier import ModifierProp
+from sparseml.pytorch.optim.modifier import PyTorchModifierYAML
+from sparseml.pytorch.sparsification.pruning.scorer import PruningParamsGradScorer
+from sparseml.pytorch.sparsification.pruning.mask_creator import (
+    PruningMaskCreator,
+    UnstructuredPruningMaskCreator,
+)
+from sparseml.pytorch.sparsification.pruning.modifier_pruning_base import (
+    BaseGradualPruningModifier,
+)
+from sparseml.utils import ALL_TOKEN
+__all__ = [
+    "MFACPruningModifier",
+    "MFACPruningParamsScorer",
+    "FisherInverse",
+    "FisherInverseFast",
+    "FisherInverseFastBlock",
+    "FisherInverseFastPageSwap",
+    "FisherInverseFastSmallBlocks",
+]
+
+_LOGGER = logging.getLogger(__name__)
+BYTES_IN_MIB = 1024 ** 2
+@PyTorchModifierYAML()
+class MFACPruningModifier(BaseGradualPruningModifier):
```

Comment:
consider inheriting from `GMPruningModifier` since the main change is just the scorer, then we get the ModifierProps for free